app/models/server.rb

```diff
@@ -71,301 +66,79 @@ class Server < ActiveRecord::Base
-            dbfile = DataFile.new
-            dbfile.path = zip_path
-            dbfile.directory = dir
-            dbfile.save!
-            DataFile.update_all({:name => File.basename(zip_path)}, {:id => dbfile.id})
-          end
-          if recordable_type == "Match"
-            recordable.demo = dbfile
-            recordable.save
-          end
-        end
-      end
-
-      def make_stats
-        graph = Gruff::Line.new
-        graph.title = name
-        pings = []
-        players = []
-        labels = {}
-        n = 0
-
-        for version in versions.all(:order => "updated_at DESC", :limit => 30).reverse
-          pings << version.ping.to_i
-          players << version.players.to_i
-          labels[n] = version.updated_at.strftime("%H:%M") if n % 3 == 0
-          n = n + 1
-        end
-
-        graph.theme_37signals
-        graph.data("Ping", pings, '#057fc0')
-        graph.data("Players", players, '#ff0000')
-        graph.labels = labels
-        graph.write(graphfile)
-      end
-
-      def default_record
-        #		if self.default
-        #			rcon_exec "record demos/auto-" + Verification.uncrap(default.name)
-        #			rcon_exec "serverpassword " + default.password
-        #			rcon_exec "connect " + default.addr
-        #		end
-end
-
-def is_free time
-  challenges.around(time).pending.count == 0 and matches.around(time).count == 0
-end
-
-def can_create? cuser
-  cuser
-end
+  has_many :logs
+  has_many :matches
+  has_many :challenges
+  belongs_to :user
+  belongs_to :recordable, :polymorphic => true
+
+  before_create :set_category
+
+  acts_as_versioned
+  non_versioned_columns << 'name'
+  non_versioned_columns << 'description'
+  non_versioned_columns << 'dns'
+  non_versioned_columns << 'ip'
+  non_versioned_columns << 'port'
+  non_versioned_columns << 'rcon'
+  non_versioned_columns << 'password'
+  non_versioned_columns << 'irc'
+  non_versioned_columns << 'user_id'
+  non_versioned_columns << 'official'
+  non_versioned_columns << 'domain'
+  non_versioned_columns << 'reservation'
+  non_versioned_columns << 'recording'
+  non_versioned_columns << 'idle'
+  non_versioned_columns << 'default_id'
+  non_versioned_columns << 'active'
+  non_versioned_columns << 'recordable_type'
+  non_versioned_columns << 'recordable_id'
+
+  def domains
+    {DOMAIN_HLTV => "HLTV", DOMAIN_HLDS => "NS Server", DOMAIN_NS2 => "NS2 Server"}
+  end
 
-def can_update? cuser
-  cuser and cuser.admin? or user == cuser
-end
+  def to_s
+    name
+  end
 
-def can_destroy? cuser
-  cuser and cuser.admin?
-end
+  def addr
+    ip + ":" + port.to_s
+  end
 
-def self.refresh
-  servers = ""
-  Server.hlds.active.all.each do |server|
-    servers << " -a2s " + server.ip + ":" + server.port.to_s
+  def set_category
+    self.category_id = (domain == DOMAIN_NS2 ? 45 : 44 )
   end
 
-  file = File.join(Rails.root, TMPFILE)
-  system "#{QSTAT} -xml #{servers} | grep -v '<name>' > #{file}"
+  def is_free time
```

Comment:
Use def with parentheses when there are parameters.<br>Rename `is_free` to `free?`.